tasks/src/main/java/ru/sberbank/school/task02/calculator/ExtendedCurrencyCalculator.java

```diff
@@ -0,0 +1,91 @@
+            return Optional.empty();
+        }
+        showQuote(quote.get());
+        if (operation == ClientOperation.SELL) {
+                return Optional.of(BigDecimal.valueOf(1).divide(quote.get().getBid(),10, RoundingMode.HALF_UP));
+        }
+        if (operation == ClientOperation.BUY) {
+            return Optional.of(BigDecimal.valueOf(1).divide(quote.get().getOffer(), 10, RoundingMode.HALF_UP));
+        }
+        System.out.println("Return Optional.empty()");
+        return Optional.empty();
+    }
+
+    Optional<Quote> findQuote(Comparator<Quote> comparator, List<Quote> quotes, ClientOperation operation) {
+        List<Quote> finalQuoteList = new ArrayList<>();
+        BigDecimal countCurr, volumeInCurr;
+        for (Quote quote: quotes) {
+            if (operation == ClientOperation.SELL) {
+                countCurr = quote.getBid();
+            } else {
+                countCurr = quote.getOffer();
+            }
+            volumeInCurr = quote.getVolumeSize().multiply(countCurr).setScale(10, RoundingMode.HALF_UP);
+            if (quote.getVolume().isInfinity() || volumeInCurr.compareTo(amountOfRequest) > 0) {
+                finalQuoteList.add(quote);
+            }
+        }
+        if (finalQuoteList.size() > 0) {
+            finalQuoteList.sort(comparator);
+            showQuotes(finalQuoteList);
+            return Optional.of(finalQuoteList.get(0));
+        }
+        return Optional.empty();
+    }
+
+    @Override
+    public Optional<BigDecimal> convertReversed(ClientOperation operation, Symbol symbol, BigDecimal amount, Beneficiary beneficiary) {
+        quotes = externalQuotesService.getQuotes(Symbol.USD_RUB);
+
+        if (operation == null) {
+            throw new NullPointerException("operation is null!");
+        }
+        if (beneficiary == null) {
+            throw new NullPointerException("beneficiary is null!");
+        }
+
+        this.symbolOfRequest = symbol;
+        this.amountOfRequest = amount;
+        System.out.println("Get benificiary: " + beneficiary.toString());
+        System.out.println("Get symbol: " + symbol.getSymbol());
+        System.out.println("Get client operation: " + operation.toString());
+        System.out.println("Get request volume: " +  amountOfRequest );
+        showQuotes(quotes);
+
+        if (check()) {
+            return extendedOperation(operation, beneficiary, quotes);
+        }
+        System.out.println("Return Optional.empty() after check");
+        return Optional.empty();
+    }
+
+    @Override
+    public Optional<BigDecimal> convertReversed(ClientOperation operation,
+                                                Symbol symbol,
+                                                BigDecimal amount,
+                                                double delta,
+                                                Beneficiary beneficiary) {
+         return Optional.empty();
```

Comment:
&#39;method def&#39; child has incorrect indentation level 9, expected level should be 8.

*<this is a auto generated comment from violation-comments-lib F7F8ASD8123FSDF>* *<a161562193>*